homeassistant/components/nexia/util.py

```diff
@@ -1,5 +1,22 @@
 """Utils for Nexia / Trane XL Thermostats."""
 
+from homeassistant.const import (
+    HTTP_BAD_REQUEST,
+    HTTP_INTERNAL_SERVER_ERROR,
+    HTTP_NOT_FOUND,
+)
+
+
+def is_invalid_auth_code(http_status_code):
```

Comment:
I know you've extracted this check from the existing code, but I wonder if we should just check for 401 (Forbidden) and 403 (Unauthorized) ?

Reply:
Having a 400 (bad request) be treated like unauthorized seems weird. (but their API might be weird)

Reply:
That sounds more sane.  Their api is weird with the 404s but we can treat that as cannot connect and it doesn't much matter.  Will add a test for that as well